src/SourceLocator/Type/AutoloadSourceLocator.php

```diff
@@ -95,13 +94,12 @@ class_exists($className);
      * nothing so throw an exception.
      *
      * @param string $functionName
-     * @return string
-     * @throws FunctionUndefined
+     * @return string|null
      */
     private function locateFunctionByName($functionName)
     {
         if (!function_exists($functionName)) {
-            throw new FunctionUndefined('Function ' . $functionName . ' was not already defined');
+            return null;
```

Comment:
Also after this block: https://3v4l.org/vJkTR

```php
(new ReflectionFunction('preg_match'))->getFileName()
```

This produces `false`, not `null`. Can you factor it in?